rasa/core/channels/slack.py

```diff
@@ -226,7 +192,58 @@ def _sanitize_user_message(text, uids_to_remove):
             ]:
                 text = re.sub(regex, replacement, text)
 
-        return text.rstrip().lstrip()  # drop extra spaces at beginning and end
+        return text.strip()
+
+    @staticmethod
+    def _is_interactive_message(payload):
+        """Check wheter the input is a supported interactive input type."""
+
+        supported = [
+            "button",
+            "select",
+            "static_select",
+            "external_select",
+            "conversations_select",
+            "users_select",
+            "channels_select",
+            "overflow",
+            "datepicker",
+        ]
+        if payload.get("actions"):
+            action_type = payload["actions"][0].get("type")
+            if action_type in supported:
+                return True
+            elif action_type:
+                logger.warning(
+                    "Received input from an interactive component of type "
+                    + "'{}', for which payload parsing is not yet supported.".format(
+                        payload["actions"][0]["type"]
+                    )
+                )
+        return False
+
+    @staticmethod
+    def _get_interactive_repsonse(action):
+        """Parse the payload for the response value."""
+
+        if action["type"] == "button":
+            return action.get("value")  # No input payload for link buttons
+        elif action["type"] == "select":
+            return action["selected_options"][0]["value"]
```

Comment:
this access looks dangerous, are all of these always present? `["selected_options"][0]["value"]`

Reply:
This is pulling from slack's response, not anything the user wrote, so as long as the slack api sends the right thing, yes. But we can still make it safer since this is the old version of select menus anyway, who knows how long they will support  `"select"` as opposed to `"static_select"`

Reply:
i'm not sure what happens if the bot-writer doesn't include a `"value"` for the menu option in their list, but I assume the posting of the menu won't go through 🤔